src/status_im/contexts/profile/settings/screens/password/change_password/new_password_form.cljs

```diff
@@ -0,0 +1,161 @@
+        :icon-color (when (= status :default)
+                      colors/white-70-blur)}
+       text])]])
+
+(defn- calc-password-strength
+  [validations]
+  (->> (vals validations)
+       (filter true?)
+       (count)))
+
+(defn- help
+  [{:keys [validations]}]
+  (let [{:keys [lower-case? upper-case? numbers? symbols?]} validations
+        password-strength                                   (calc-password-strength validations)]
+    [rn/view
+     [quo/strength-divider {:type (strength-status password-strength :info)}
+      (i18n/label :t/password-creation-tips-title)]
+     [rn/view {:style style/password-tips}
+      [quo/tips {:completed? lower-case?}
+       (i18n/label :t/password-creation-tips-1)]
+      [quo/tips {:completed? upper-case?}
+       (i18n/label :t/password-creation-tips-2)]
+      [quo/tips {:completed? numbers?}
+       (i18n/label :t/password-creation-tips-3)]
+      [quo/tips {:completed? symbols?}
+       (i18n/label :t/password-creation-tips-4)]]]))
+
+(defn- password-validations
+  [password]
+  (let [validations (juxt utils.string/has-lower-case?
+                          utils.string/has-upper-case?
+                          utils.string/has-numbers?
+                          utils.string/has-symbols?)]
+    (->> password
+         (validations)
+         (zipmap constant/password-tips))))
+
+(defn view
+  []
+  (let [customization-color                            (rf/sub [:profile/customization-color])
+        [password set-password]                        (rn/use-state "")
+        [repeat-password set-repeat-password]          (rn/use-state "")
+        [disclaimer-accepted? set-disclaimer-accepted] (rn/use-state false)
+        [focused? set-focused]                         (rn/use-state false)
+        [show-validation? set-show-validation]         (rn/use-state false)
+
+        ;; validations
+        validations                                    (password-validations password)
+        long-enough?                                   (utils.string/at-least-n-chars? password 10)
+        empty-password?                                (empty? password)
+        same-passwords?                                (and (not empty-password?)
+                                                            (= password repeat-password))
+        meet-requirements?                             (and (not empty-password?)
+                                                            long-enough?
+                                                            same-passwords?
+                                                            disclaimer-accepted?)
+        error?                                         (and show-validation?
+                                                            (not same-passwords?)
+                                                            (not empty-password?))
+
+        ;; handlers
+        on-change-password                             (fn [new-value]
```

Comment:
A lot of `rn/use-callback` calls can be added here ;)

Reply:
not sure all of them should be memoised. Most of the re-rendering in this screen will happen due to input value changes and these handlers will be re-created anyways. That's why I only memoised the ones that don't depend on the input values.

Reply:
As far as I understand, use/callback allows you to avoid re-creating functions. You can specify the value as a dependency and its change will be taken into account

Reply:
yes, but the functions will be re-created and memoised again when the dependencies change. So with an input value as a dependency, it will still re-create the functions and will memoise them on every input change.